src/components/Composer/index.js

```diff
@@ -308,52 +318,18 @@ class Composer extends React.Component {
             const domparser = new DOMParser();
             const embeddedImages = domparser.parseFromString(pastedHTML, TEXT_HTML).images;
 
-            // If HTML has img tag, then fetch images from it.
+            // If HTML has img tag, then we only paste the plain text. This is because
+            // fetching the image via fetch triggers a Content-Security-Policy error.
             if (embeddedImages.length > 0 && embeddedImages[0].src) {
-                // If HTML has emoji, then treat this as plain text.
-                if (embeddedImages[0].dataset && embeddedImages[0].dataset.stringifyType === 'emoji') {
-                    const plainText = event.clipboardData.getData('text/plain');
-                    this.paste(plainText);
-                    return;
-                }
-                fetch(embeddedImages[0].src)
-                    .then((response) => {
-                        if (!response.ok) {
-                            throw Error(response.statusText);
-                        }
-                        return response.blob();
-                    })
-                    .then((x) => {
-                        const extension = IMAGE_EXTENSIONS[x.type];
-                        if (!extension) {
-                            throw new Error(this.props.translate('composer.noExtensionFoundForMimeType'));
-                        }
-
-                        return new File([x], `pasted_image.${extension}`, {});
-                    })
-                    .then(this.props.onPasteFile)
-                    .catch(() => {
-                        const errorDesc = this.props.translate('composer.problemGettingImageYouPasted');
-                        Growl.error(errorDesc);
-
-                        /*
-                         * Since we intercepted the user-triggered paste event to check for attachments,
-                         * we need to manually set the value and call the `onChangeText` handler.
-                         * Synthetically-triggered paste events do not affect the document's contents.
-                         * See https://developer.mozilla.org/en-US/docs/Web/API/Element/paste_event for more details.
-                         */
-                        this.handlePastedHTML(pastedHTML);
-                    });
+                this.handlePastePlainText(event);
```

Comment:
@parasharrajat If you read my proposal carefully, I explain the caveat of why we should paste plaintext instead of pasting the HTML (https://github.com/Expensify/App/issues/18583#issuecomment-1598515619)

If any images have links attached, and we try to paste them, it leads to formatting issues in the markdown that is generated.

Please let me know what is the desired behaviour in this instance as I thought you already approved my proposal.

Reply:
This is what happens when I set it to paste HTML:

https://github.com/Expensify/App/assets/31749903/7c9313d6-36b9-4348-8e44-81b12a186c0d

Reply:
@parasharrajat This is only when the images are wrapped in a link.

Reply:
> It is strange that image tags are converted to links
It would be because the image are used as links? @qispark Can you test the same on Slack and WhatsApp and see what is being pasted there? But I agree with @parasharrajat that if the copied content has HTML then we would want to paste them as HTML (MINUS the image part).

Reply:
@techievivek The images are stripped on platforms like Slack.

The only way forward then is to update ExpensiMark, as it is formatting the pasted content into Markdown. I also mentioned this in my original comment (https://github.com/Expensify/App/issues/18583#issuecomment-1598515619).

I think if we can implement a new function in ExpensiMark, which does the same thing (e.g. `parser.htmlToMarkdownWithoutImages`) we can utilise it here. What are your thoughts, and is this a feasible option?

Reply:
@parasharrajat Let me know if I should action on anything based on your feedback.

Reply:
@qispark Based on Rajat's comment,` If images are wrapped in a link then it is fine if they are parsed as links in the pasted code.` I think the image was wrapped in a link in the recording above, right? (Can you confirm this by inspecting that element?) and when we are pasting it is being rendered as links.

@parasharrajat
> But if the image itself is being converted to links then that is a problem

I don't _think_ that is happening in this case. @qispark can you please confirm this?


And now for the last part

> Why are we not pasting the HTML? embeddedImages[0].dataset && embeddedImages[0].dataset.stringifyType === 'emoji' require pasting text but other than this we can use html pasting.

@qispark What do you think can be done to fix this? Do we need to update the ExpensiMark logic? I think that is very much doable, so if that's the case please push a PR in the expensify-common repo and let's keep this moving.